corehq/apps/locations/tests/test_api_resources.py

```diff
@@ -409,3 +414,55 @@ def test_patch_list_missing_location_id(self):
         self.assertEqual(response.json(),
                          {'error': "Could not update: could not find location with"
                                    f" given ID {unknown_location_id} on the domain."})
+
+
+class TestV0_6LocationsList(APIResourceTest):
+    api_name = 'v0.6'
+    resource = v0_6.LocationResource
+    location_type_names = ['state', 'county', 'city']
+    location_structure = [
+        ('Massachusetts', [
+            ('Middlesex', [
+                ('Cambridge', []),
+                ('Somerville', []),
+            ]),
+            ('Suffolk', [
+                ('Boston', []),
+            ])
+        ])
+    ]
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        with freeze_time(datetime(2025, 1, 1)):
+            cls.location_types, cls.locations = setup_locations_and_types(
+                cls.domain.name,
+                cls.location_type_names,
+                [],
+                cls.location_structure,
+            )
+            cls.locations['Middlesex'].location_id = 'middlesex_uuid'
+            cls.locations['Middlesex'].save()
+
+        with freeze_time(datetime(2025, 4, 1)):
+            cls.locations['Boston'].save()
+            cls.locations['Cambridge'].save()
+
+    @generate_cases([
+        ('site_code=boston', ['boston']),
+        ('last_modified.gte=2025-03-31', ['boston', 'cambridge']),
+        ('last_modified.gte=2020-01-01&last_modified.lt=2025-03-31', [
+            'massachusetts', 'middlesex', 'somerville', 'suffolk',
+        ]),
+        ('last_modified.gt=2025-03-31&last_modified.lte=2025-03-30', []),
+        ('location_type_code=county', ['middlesex', 'suffolk']),
+        ('parent_location_id=middlesex_uuid', ['cambridge', 'somerville']),
+    ])
+    def test_api_filters(self, querystring, expected):
+        url = f"{self.list_endpoint}?{querystring}"
+        response = self._assert_auth_get_resource(url)
+        self.assertEqual(response.status_code, 200)
+        self.assertItemsEqual([
+            loc['site_code'] for loc in response.json()['objects']
```

Comment:
I love that we have new tests here, but I just want to mention that I found them a bit hard to understand. If one of these tests cases were to fail, I'd need to read and understand the entire test suite in order to understand why, because all the context is in the setup method. I think tests that contain their context directly in the test body tend to be easier to understand. I'm imagining pseudo-code like the following would have also worked:
```
def test_modified_comparisons(self):
  # set up location A, created on date x
  # set up location B, created on date y
  # execute api filtering to include y but not x
  # verify that the set of locations returned only includes B
```

Reply:
Doing the setup separately allows me to do it once and share between test cases - this test class is just a setup function and this test function with the different use-cases.  Running this test file takes 22 seconds locally, so I didn't want to balloon that too much.